source/simulator/core.cc

```diff
@@ -1976,6 +1976,26 @@ namespace aspect
           break;
         }
 
+        case NonlinearSolver::Binary_input:
+        {
+          // Identical to IMPES except does not solve Stokes equation
+          if (parameters.free_surface_enabled)
+            free_surface->execute ();
+       
+          std::string fileName = parameters.input_directory + "/" + "solution-" + Utilities::int_to_string(timestep_number, 5) + ".mesh";
+          triangulation.load(fileName.c_str());
+          parallel::distributed::SolutionTransfer<dim, LinearAlgebra::BlockVector> sol_trans(dof_handler);
+          sol_trans.deserialize (solution);
+          //triangulation.clear();
```

Comment:
> Therefore, I ended up adding a new Nonlinear solver called binary_input to circumvent these issues. Is this the proper place for this feature or is the extension of the prescribed stokes plugin the right approach?

I think we need to restructure the solvers at some point anyway to be more flexible it what to iterate when, etc.. But today is not that day. :-)

Is your binary-input for Stokes only, Stokes+Temperature+Composition, or either? It probably makes the most sense to read in all variables, right? If yes, "prescribed Stokes" is obviously the wrong place. If you only want to load the Stokes solution, we should try to fit it in there even if it means we need to extend/change the interface. You could, for example, add an additional function to the interface like ``void fill_vector(MPI::Vector &solution, ...)`` with an empty default implementation. This would be called once in the beginning instead of interpolating the function.